src/harness/resources/3d_frag.glsl

```diff
@@ -4,24 +4,28 @@
 in vec3 v_frag_pos;
 in vec3 v_normal;
 in vec2 v_tex_coord;
+layout(origin_upper_left) in vec4 gl_FragCoord;
```

Comment:
I had to add:
```
#extension GL_ARB_fragment_coord_conventions: require
```
otherwise the highlighted change prevents the shader from compiling:
```
[INFO] GLRenderer_Init OpenGL vendor string: AMD
[INFO] GLRenderer_Init OpenGL renderer string: AMD Radeon Vega 11 Graphics (raven, LLVM 14.0.6, DRM 3.49, 6.1.4-arch1-1)
[INFO] GLRenderer_Init OpenGL version string: 4.6 (Compatibility Profile) Mesa 22.3.2
[INFO] GLRenderer_Init OpenGL shading language version string: 4.60
[PANIC] CreateShaderProgram shader 3d failed to compile: 0:9(8): error: unrecognized layout identifier `origin_upper_left'
```

Reply:
thanks for the report! I'll have to do some digging - that extension is not supported on my main dev system (OSX).

Reply:
@zear could you check this branch now compiles (and looks correct) for you?

Reply:
Hijacking this thread because I have this error:
```
[INFO] GLRenderer_Init OpenGL vendor string: Intel
[INFO] GLRenderer_Init OpenGL renderer string: Mesa Intel(R) HD Graphics 4600 (HSW GT2)
[INFO] GLRenderer_Init OpenGL version string: 4.6 (Core Profile) Mesa 22.1.7
[INFO] GLRenderer_Init OpenGL shading language version string: 4.60
[PANIC] CreateShaderProgram shader 3d failed to compile: 0:7(9): error: `gl_FragCoord' redeclared
```
Commenting out the `in vec4 gl_FragCoord;` line made it work.

Reply:
thanks! Its great having a few different systems to test these things with. Now that we aren't overriding the layout, I don't _need_ to redeclare it, but my opengl was quite happy with it. Will remove 👍

Reply:
I confirm the same result as @madebr. Removing `in vec4 gl_FragCoord;` makes it work for me.